solutions/project/project-sandbox/network/nat.yaml

```diff
@@ -0,0 +1,79 @@
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeRouterNAT
+metadata:
+  name: project-id-nane1-nat # kpt-set: ${project-id}-nane1-nat
+  namespace: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/networking/ComputeRouter/project-id-nane1-router # kpt-set: compute.cnrm.cloud.google.com/namespaces/networking/ComputeRouter/${project-id}-nane1-router
+spec:
+  resourceID: nane1-nat # kpt-set: nane1-nat
+  natIpAllocateOption: AUTO_ONLY
+  region: northamerica-northeast1
+  routerRef:
+    name: project-id-nane1-router # kpt-set: ${project-id}-nane1-router
+  sourceSubnetworkIpRangesToNat: ALL_SUBNETWORKS_ALL_IP_RANGES
+---
+# Cloud Router northamerica-northeast1
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeRouter
+metadata:
+  name: project-id-nane1-router # kpt-set: ${project-id}-nane1-router
+  namespace: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/project-id-global-vpc1-vpc # kpt-set: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/${project-id}-global-vpc1-vpc
+spec:
+  resourceID: nane1-router # kpt-set: nane1-router
+  description: northamerica-northeast1 router
+  networkRef:
+    name: project-id-global-vpc1-vpc # kpt-set: ${project-id}-global-vpc1-vpc
+  region: northamerica-northeast1
+---
+# Cloud NAT northamerica-northeast2
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeRouterNAT
+metadata:
+  name: project-id-nane2-nat # kpt-set: ${project-id}-nane2-nat
+  namespace: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/networking/ComputeRouter/project-id-nane2-router # kpt-set: compute.cnrm.cloud.google.com/namespaces/networking/ComputeRouter/${project-id}-nane2-router
+spec:
+  resourceID: nane2-nat # kpt-set: nane2-nat
+  natIpAllocateOption: AUTO_ONLY
+  region: northamerica-northeast2
+  routerRef:
+    name: project-id-nane2-router # kpt-set: ${project-id}-nane2-router
+  sourceSubnetworkIpRangesToNat: ALL_SUBNETWORKS_ALL_IP_RANGES
+---
+# Cloud Router northamerica-northeast2
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeRouter
+metadata:
+  name: project-id-nane2-router # kpt-set: ${project-id}-nane2-router
+  namespace: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/project-id-global-vpc1-vpc # kpt-set: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/${project-id}-global-vpc1-vpc
+spec:
+  resourceID: nane2-router # kpt-set: nane2-router
```

Comment:
Again, maybe missing `${project-id}` inside the setter.

Reply:
you are right luc, I thought it was a comment for the other line. When resource id is not specified, the resource name in gcp will have the same value as the metadata name. I'm setting a very complex metadata name to ensure uniqueness accross the k8s namespace but the resource name only need to be unique within it's project so I'm defining a much simpler resource name. As you pointed out, I will remove the kpt-set because it is not required.